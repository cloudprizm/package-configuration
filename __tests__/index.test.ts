import fs from 'fs'
jest.mock('fs')

import {
  coerceVersions,
  getLocalPackageJson,
  getDependencyVersions,
  persistVersionsFromPackage
} from '../src/index'

test('get deps versions', () => {
  const pkg = getLocalPackageJson('../package.json')
  const originalVersions = getDependencyVersions(pkg)
  const cdnVersions = coerceVersions(originalVersions)

  // yeah yeah laziness
  expect(cdnVersions).toMatchSnapshot()
})

test('save version to file - if all ok', () => {
  const writeFile = jest
    .fn()
    .mockResolvedValue('all good')

  // @ts-ignore
  fs.promises = {
    writeFile
  }

  const config = {
    packagePath: '../package.json',
    destinationPath: 'versions.json'
  }

  return persistVersionsFromPackage
    .run(config)
    .then(d => {
      if (d.isRight()) {
        expect(writeFile).toHaveReturned()
        expect(writeFile).toHaveBeenCalledWith(config.destinationPath, d.value)
      } else fail('should be right')
    })
})

test('save version to file - if there is a problem', () => {
  const errorMessage = 'ulala sth is wrong'
  const writeFile = jest
    .fn()
    .mockRejectedValue(new Error(errorMessage))

  // @ts-ignore
  fs.promises = {
    writeFile
  }

  const config = {
    packagePath: '../package.json',
    destinationPath: 'versions.json'
  }

  return persistVersionsFromPackage
    .run(config)
    .then(d => {
      if (d.isLeft()) expect(d.value).toBe(errorMessage)
      else fail('should be left')
    })
})
