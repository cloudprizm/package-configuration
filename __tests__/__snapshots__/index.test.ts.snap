// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get deps versions 1`] = `
Object {
  "@babel/cli": "7.1.2",
  "@babel/core": "7.1.2",
  "@hungry/babel-preset-cli": "0.1.1",
  "@types/cosmiconfig": "5.0.3",
  "babel-core": "7.0.0",
  "babel-jest": "23.6.0",
  "concurrently": "4.0.1",
  "cosmiconfig": "5.0.6",
  "fp-ts": "1.10.0",
  "jest": "23.6.0",
  "ramda": "0.25.0",
  "semver": "5.6.0",
  "typescript": "3.1.3",
}
`;
